### PR TITLE
config.tf: add "none" networking option to config

### PR DESCRIPTION
**Pull request by spangenberg**

### CODE REVIEW

File: modules/tectonic/assets.tf [hunk 1]
```diff
@@ -101,9 +108,7 @@ resource "template_dir" "tectonic" {
     image_re            = "${var.image_re}"
     kube_dns_service_ip = "${cidrhost(var.service_cidr, 10)}"
 
-    tectonic_networking = "${var.tectonic_networking}"
-    calico_mtu          = "${var.calico_mtu}"
-    cluster_cidr        = "${var.cluster_cidr}"
+    network_config = "${indent(6, local.network_config_yaml)}"
```
Comment (alexsomesan) [inline modules/tectonic/assets.tf]: Very fancy 👍

File: modules/tectonic/assets.tf [hunk 2]
```diff
@@ -1,3 +1,10 @@
+locals {
+  calico_config_yaml   = "${var.tectonic_networking == "calico" ? "calicoConfig:\n    mtu: ${var.calico_mtu}" : ""}"
+  network_config_yaml  = "${join("\n", compact(list(local.network_profile_yaml, local.pod_cidr_yaml, local.calico_config_yaml)))}"
```
Comment (alexsomesan) [inline modules/tectonic/assets.tf]: I presume the `compact()` ensures no empty lines when some attributes are missing, right?

Comment (spangenberg) [reply modules/tectonic/assets.tf]: Exactly

File: modules/tectonic/assets.tf (hunk 1 above)
Comment (spangenberg) [reply modules/tectonic/assets.tf]: :smirk:

**Pull request merged**